demo/demo.h

```diff
@@ -37,37 +44,207 @@ DECLARE_int32(demo_port);
 namespace demo {
 
 using bricks::FileSystem;
+using bricks::time::Now;
+using bricks::strings::Printf;
 using bricks::net::api::HTTP;
 using bricks::net::api::Request;
 using bricks::net::HTTPResponseCode;
 using namespace bricks::gnuplot;
+using namespace bricks::cerealize;
+
+
+struct LayoutCell {
+  std::string meta_url = "/meta";
+
+  // Define only output serialization (`JSON.stringify()`), forbid input serialization (`JSON.parse()`).
+  template <typename A>
+  void save(A& ar) const {
+    ar(CEREAL_NVP(meta_url));
+  }
+};
+static_assert(is_write_cerealizable<LayoutCell>::value, "");
+static_assert(!is_read_cerealizable<LayoutCell>::value, "");
+
+struct LayoutItem {
+  std::vector<LayoutItem> row;
+  std::vector<LayoutItem> col;
+  LayoutCell cell;
+
+  // Define only output serialization (`JSON.stringify()`), forbid input serialization (`JSON.parse()`).
+  template <typename A>
+  void save(A& ar) const {
+    if (!row.empty()) {
+      ar(CEREAL_NVP(row));
+    } else if (!col.empty()) {
+      ar(CEREAL_NVP(col));
+    } else {
+      ar(CEREAL_NVP(cell));
+    }
+  }
+};
+static_assert(is_write_cerealizable<LayoutItem>::value, "");
+static_assert(!is_read_cerealizable<LayoutItem>::value, "");
+
+struct ExampleMeta {
+  struct Options {
+    std::string header_text = "Real-time Data Made Easy";
+    std::string color = "blue";
+    double min = 0;
+    double max = 1;
+    double time_interval = 10000;
+    template <typename A>
+    void serialize(A& ar) {
+      ar(CEREAL_NVP(header_text),
+         CEREAL_NVP(color),
+         CEREAL_NVP(min),
+         CEREAL_NVP(max),
+         CEREAL_NVP(time_interval));
+    }
+  };
+  std::string data_url = "/data";
+  std::string visualizer_name = "plot-visualizer";
+  Options visualizer_options;
+  template <typename A>
+  void serialize(A& ar) {
+    ar(CEREAL_NVP(data_url), CEREAL_NVP(visualizer_name), CEREAL_NVP(visualizer_options));
+  }
+};
+
 
-struct DemoServer {
+class DemoServer {
+ public:
   DemoServer(int port = FLAGS_demo_port) : port_(port) {
+    printf("Preparing to listen on port %d...\n", port_);
```

Comment:
I've copy-pasted from below where `printf` was used. Will change to `Printf` from `Bricks`.

I needed this one to prove the server fails on `FileSystem::ScanDir` when the directory does not exist. We are likely to have `try..catch` around it with more readable error output.